node/src/components/linear_chain_sync.rs

```diff
@@ -952,3 +957,36 @@ pub(crate) fn clean_linear_chain_state(
     let key = create_state_key(&chainspec);
     storage.del_state_store(key)
 }
+
+async fn execute_finalized_block<REv>(
```

Comment:
This is duplicated in the era supervisor, is it? We can deduplicate this if we move it to e.g. `utils` — or some new `common` module for this type of logic?

Reply:
Well, `linear_chain_sync` is going away so I didn't think having this duplicated code was a big deal.